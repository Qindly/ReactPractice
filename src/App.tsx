import { useState, useEffect } from "react";
import UserProfile from "./components/UserProfile";
import LoginForm from "./components/LoginForm";
import UserContext from "./utils/MyContext/UserContext";
import { useFetchUser } from "./utils/Myhooks/useFetchUser";
//这里是路由的出口
import { Outlet, Link } from "react-router-dom";
const App = () => {
  const { userData, loading, error } = useFetchUser(1);
  console.log("userData:", userData);
  console.log("loading:", loading);
  console.log("error:", error);
  const [users, setUsers] = useState([
    { id: 1, username: "Apple", email: "Apple.email", displayName: "AAAApple" },
    { id: 1, username: "Apple", email: "Apple.email", displayName: "AAAApple" },
  ]);
  //记住如果用花括号，就要显性返回，即要有return，没有return的话要用圆括号()
  return (
    <>
      {users.map((user) =>
        <UserProfile key={user.id} user={user} setUsers={setUsers} />
      )}
    </>
  );
};

export default App;
{
  /* <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/users">Users</Link>
          </li>
          <li>
            <Link to="/blog">Blog</Link>
          </li>
        </ul>
      </nav> */
}
{
  /* <UserContext.Provider
        value={{
          id: 1,
          username: "Apple",
          email: "Apple.email",
          displayName: "AAAApple",
        }}
      >
        <div>11</div>
        <Outlet />
      </UserContext.Provider> */
}
