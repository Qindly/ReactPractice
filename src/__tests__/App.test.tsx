import { describe, expect, it, test } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../App";

describe("App", () => {
  it("should render user", () => {
    const {container}= render(<App />);
    expect(container).toMatchSnapshot();
  });
});
