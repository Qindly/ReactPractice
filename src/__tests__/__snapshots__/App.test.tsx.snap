// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render user 1`] = `
<div>
  <ul>
    <li>
      ID:
      1
    </li>
    <li>
      Username:
      Apple
    </li>
    <li>
      Email: 
      Apple.email
    </li>
  </ul>
  <button>
    modify
  </button>
  <div>
    <button>
      finish
    </button>
    <button>
      cancel
    </button>
  </div>
  <button>
    delete
  </button>
  <ul>
    <li>
      ID:
      1
    </li>
    <li>
      Username:
      Apple
    </li>
    <li>
      Email: 
      Apple.email
    </li>
  </ul>
  <button>
    modify
  </button>
  <div>
    <button>
      finish
    </button>
    <button>
      cancel
    </button>
  </div>
  <button>
    delete
  </button>
</div>
`;
